fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2469,6 +2485,91 @@ ACTOR Future<Void> reportTxnTagCommitCost(UID myID,
 	}
 }
 
+namespace {
+struct ExpireServerEntry {
+	int64_t timeReceived;
+	int expectedCount = 0;
+	int receivedCount = 0;
+	bool initialized = false;
+};
+
+struct IdempotencyKey {
+	Version version;
+	uint8_t highOrderBatchIndex;
+	bool operator==(const IdempotencyKey& other) const {
+		return version == other.version && highOrderBatchIndex == other.highOrderBatchIndex;
+	}
+};
+
+} // namespace
+
+namespace std {
+template <>
+struct hash<IdempotencyKey> {
+	std::size_t operator()(const IdempotencyKey& key) const {
+		std::size_t seed = 0;
+		boost::hash_combine(seed, std::hash<Version>{}(key.version));
+		boost::hash_combine(seed, std::hash<uint8_t>{}(key.highOrderBatchIndex));
+		return seed;
+	}
+};
+
+} // namespace std
+
+ACTOR static Future<Void> idempotencyIdsExpireServer(
+    Database db,
+    PublicRequestStream<ExpireIdempotencyIdRequest> expireIdempotencyId,
+    FutureStream<ExpireIdempotencyKeyValuePairRequest> expireIdempotencyKeyValuePair,
```

Comment:
The names of these two things (and their associated requests) were a bit unclear to me until I read through this code, and I could see them causing a little confusion. Maybe we could rename them to make it clear which one comes from the client acknowledging the commit and which one comes from the proxy reporting the commit.

Reply:
s/ExpireIdempotencyKeyValuePairRequest/ExpectedIdempotencyIdCountForKey/